compiler/rustc_hir/src/hir.rs

```diff
@@ -3182,6 +3182,20 @@ pub enum Node<'hir> {
 }
 
 impl<'hir> Node<'hir> {
+    /// Get the identifier of this `Node`, if applicable.
+    ///
+    /// # Edge cases
+    ///
+    /// Calling `.ident()` on a [`Node::Ctor`] will return `None`
+    /// because `Ctor`s do not have identifiers themselves.
+    /// Instead, call `.ident()` on the parent struct/variant, like so:
+    ///
+    /// ```ignore (illustrative)
+    /// ctor
+    ///     .ctor_hir_id()
+    ///     .and_then(|ctor_id| tcx.hir().find(tcx.hir().get_parent_node(ctor_id)))
+    ///     .and_then(|parent| parent.ident())
+    /// ```
```

Comment:
Should we make this part of the call to `ident`, requiring people pass in `tcx` so they don't have to worry?

Reply:
I don't think we have access to `rustc_middle` from `rustc_hir` because `rustc_middle` depends on `rustc_hir`. If you have a suggestion for how to do it, I'm happy to, but it might not be worth it for now.

Reply:
Ok, fair enough. Don't worry about it :)